frontend/src/pages/ProductPage.js

```diff
@@ -1,19 +1,50 @@
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
 import { useParams } from 'react-router-dom';
-import products from '../products';
 import MainContent from '../components/MainContent';
 import ProductDescription from '../components/ProductDescription';
+import Loader from 'react-loader-spinner';
 
 const ProductPage = () => {
   const { id } = useParams();
-  const product = products.find(p => p._id === id);
+  const [product, setProduct] = useState({});
+  const url = 'api/v1/products/';
+
+  axios.defaults.baseURL = 'http://localhost:8000';
```

Comment:
This feels like the wrong place for this. I would create a `src/api.js` file and put this kind of stuff there.

```js
export const baseUrl = 'http://localhost:8000';
export const products = 'api/v1/products/';

axios.defaults.baseURL = baseUrl;
```

And then you could import those from there:

```js
import api from '../../api';

# and then use it like

const { data } = await axios.get(`${api.products}${id}`);